### PR TITLE
feat(#90): Refactor egress example to be integrated with tutorial 

### PR DESCRIPTION
**Pull request by lordofthejars**

### CODE REVIEW

File: recommendation/java/vertx/src/main/java/com/redhat/developer/demos/recommendation/RecommendationVerticle.java [hunk 1]
```diff
@@ -2,16 +2,21 @@
...
 import io.vertx.ext.web.Router;
 import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.codec.BodyCodec;
 
 public class RecommendationVerticle extends AbstractVerticle {
 
-    private static final String RESPONSE_STRING_FORMAT = "recommendation v1 from '%s': %d\n";
+    private static final String RESPONSE_STRING_FORMAT = "recommendation v3 from '%s': %d\n";
```
Comment (rafabene) [inline recommendation/java/vertx/src/main/java/com/redhat/developer/demos/recommendation/RecommendationVerticle.java]: You committed the file as v3. It should be v1 until we tell users to change it.

File: docs/8egress.adoc [hunk 2]
```diff
@@ -1,174 +1,165 @@
...
+oc start-build recommendation-v3 --from-dir=. --follow
+----
 
-oc expose service egressgithub
+==== Wait for v3 to be deployed
 
-curl egressgithub-istioegress.$(minishift ip).nip.io
+Wait for those pods to show "2/2", the istio-proxy/envoy sidecar is part of that pod
 
-cd ../..
```
Comment (rafabene) [inline docs/8egress.adoc]: This is not being formatted correctly

File: docs/8egress.adoc [hunk 3]
```diff
@@ -1,174 +1,165 @@
...
-
-curl localhost:8080
 
-curl httpbin.org/user-agent
+Then access to the service:
 
-curl httpbin.org/headers
+[source, bash]
+----
+curl customer-tutorial.$(minishift ip).nip.io
```
Comment (rafabene) [inline docs/8egress.adoc]: Add a note that it might take more than 10 seconds to reply with "upstream request timeout"

File: docs/8egress.adoc [hunk 4]
```diff
@@ -1,174 +1,165 @@
...
 
-curl egressgithub-istioegress.$(minishift ip).nip.io
-----
 
 Clean up
 
 [source,bash]
 ----
-iistioctl delete serviceentry httpbin-egress-rule google-egress-rule github-egress-rule -n istioegress
+iistioctl delete -f istiofiles/service-entry-egress-httpbin.yml -n tutorial
```
Comment (rafabene) [inline docs/8egress.adoc]: typo ii

File: docs/8egress.adoc (hunk 3 above)
Comment (rafabene) [reply docs/8egress.adoc]: Maybe you can add a curl -m 5 or -m 10 to specify a timeout in seconds. Having the endpoint without an answer for so long made me wonder if it was something wrong.

**Comment by lordofthejars** [comment]
@rafabene Done.

**Pull request merged**